File: 1395/brute_force.cpp
#include <cassert>
#include <iostream>
#include <vector>

using namespace std;

class Solution {
 public:
  int numTeams(vector<int>& rating) {
    size_t size = rating.size();
    int ret = 0;
    for (size_t i = 0; i < size - 2; ++i) {
      for (size_t j = i + 1; j < size - 1; ++j) {
        for (size_t k = j + 1; k < size; ++k) {
          if ((rating[i] < rating[j] && rating[j] < rating[k]) || (rating[i] > rating[j] && rating[j] > rating[k])) ret++;
        }
      }
    }
    return ret;
  }
};

int main() {
  Solution sol;
  vector<int> rating;

  rating = {2,5,3,4,1};
  assert(sol.numTeams(rating) == 3);

  rating = {2,1,3};
  assert(sol.numTeams(rating) == 0);

  rating = {1,2,3,4};
  assert(sol.numTeams(rating) == 4);
}

File: 1583/brute_force.py
from typing import List


class Solution:
    def unhappyFriends(self, n: int, preferences: List[List[int]],
                       pairs: List[List[int]]) -> int:
        paired = [0 for _ in range(n)]
        for pair in pairs:
            paired[pair[0]] = pair[1]
            paired[pair[1]] = pair[0]
        ret = 0
        for person in range(n):
            is_unhappy = False
            for prefer in preferences[person]:
                if prefer == paired[person]:
                    break
                for prefer_of_prefer in preferences[prefer]:
                    if prefer_of_prefer == paired[prefer]:
                        break
                    if prefer_of_prefer == person:
                        is_unhappy = True
                        break
            if is_unhappy:
                ret += 1
        return ret


if __name__ == '__main__':
    sol = Solution()

    n = 4
    preferences = [[1, 2, 3], [3, 2, 0], [3, 1, 0], [1, 2, 0]]
    pairs = [[0, 1], [2, 3]]
    expected = 2
    assert sol.unhappyFriends(n, preferences, pairs) == expected

    n = 2
    preferences = [[1], [0]]
    pairs = [[1, 0]]
    expected = 0
    assert sol.unhappyFriends(n, preferences, pairs) == expected

    n = 4
    preferences = [[1, 3, 2], [2, 3, 0], [1, 3, 0], [0, 2, 1]]
    pairs = [[1, 3], [0, 2]]
    expected = 4
    assert sol.unhappyFriends(n, preferences, pairs) == expected
